### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by wwylele**
I don't really want to be the one to update it, because I am neither a native speaker, nor one who knows the code style guidelines well. But it is really outdated and I am surprised no one seems to be willing to update it. So I just open this PR to find some points that are defected, to bring a discussion on them and to update it eventually.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
 
 ### General Rules
 * A lot of code was taken from other projects (e.g. Dolphin, PPSSPP, Gekko, SkyEye). In general, when editing other people's code, follow the style of the module you're in (or better yet, fix the style if it drastically differs from our guide).
-* Line width is typically 100 characters, but this isn't strictly enforced. Please do not use 80-characters.
+* Line width is typically 100 characters. Please do not use 80-characters.
 * Don't ever introduce new external dependencies into Core
 * Don't use any platform specific code in Core
 * Use namespaces often
-* Avoid the use of C-style casts and instead prefer C++-style `static_cast` and `reinterpret_cast`. Never use `const_cast` or `dynamic_cast` (we build with RTTI disabled). The only exception to this rule is for casting between two numeric types, where C-style casts are encouraged for brevity and readability.
+* Avoid the use of C-style casts and instead prefer C++-style `static_cast` and `reinterpret_cast`. Try to avoid using `dynamic_cast`. Never use `const_cast`.
```
Comment (wwylele) [inline CONTRIBUTING.md]: RTTI was re-enabled ages ago

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -59,9 +57,9 @@ char* g_some_pointer; // Pointer * and reference & stick to the type name
 
 /// A colorful enum.
 enum SomeEnum {
-    COLOR_RED,   ///< The color of fire.
-    COLOR_GREEN, ///< The color of grass.
-    COLOR_BLUE,  ///< Not actually the color of water.
+    ColorRed,   ///< The color of fire.
+    ColorGreen, ///< The color of grass.
+    ColorBlue,  ///< Not actually the color of water.
```
Comment (wwylele) [inline CONTRIBUTING.md]: I was told to do this when contributing

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,21 +1,19 @@
...
+**The issue tracker is not a support forum.** Unless you can provide precise *technical information* regarding an issue, you *should not post in it*. If you need support, first read the [FAQ](https://github.com/citra-emu/citra/wiki/FAQ) and then either visit our IRC channel, [our forum](https://discuss.citra-emu.org/) or ask in a general emulation forum such as [/r/emulation](https://www.reddit.com/r/emulation/). If you post support questions, generic messages to the developers or vague reports without technical details, they will be closed and locked.
 
 If you believe you have a valid issue report, please post text or a screenshot from the log (the console window that opens alongside Citra) and build version (hex string visible in the titlebar and zip filename), as well as your hardware and software information if applicable.
 
-While issues about specific games not booting are valid bugs, we are currently not interested in them unless you have several games which fail with the same or similar messages. (In which case please file only a single issue about it.) There are too many non-working games right now to file individual issues for every one of them.
```
Comment (wwylele) [inline CONTRIBUTING.md]: We already have lots of games working

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,21 +1,19 @@
...
 
 ### General Rules
 * A lot of code was taken from other projects (e.g. Dolphin, PPSSPP, Gekko, SkyEye). In general, when editing other people's code, follow the style of the module you're in (or better yet, fix the style if it drastically differs from our guide).
-* Line width is typically 100 characters, but this isn't strictly enforced. Please do not use 80-characters.
+* Line width is typically 100 characters. Please do not use 80-characters.
 * Don't ever introduce new external dependencies into Core
 * Don't use any platform specific code in Core
 * Use namespaces often
-* Avoid the use of C-style casts and instead prefer C++-style `static_cast` and `reinterpret_cast`. Never use `const_cast` or `dynamic_cast` (we build with RTTI disabled). The only exception to this rule is for casting between two numeric types, where C-style casts are encouraged for brevity and readability.
```
Comment (bunnei) [inline CONTRIBUTING.md]: Why was `where C-style casts are encouraged for brevity and readability.` removed?

File: CONTRIBUTING.md (hunk 2 above)
Comment (bunnei) [inline CONTRIBUTING.md]: Either has been acceptable in the past, but agree with your change since this is probably the preferred format

**Comment by bunnei** [comment]
Mostly LGTM except for the one comment

File: CONTRIBUTING.md (hunk 4 above)
Comment (wwylele) [inline CONTRIBUTING.md]: oops, didn't mean to do that

**Pull request merged**